seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/SubPlan.java

```diff
@@ -312,6 +312,10 @@ public synchronized void updatePipelineState(@NonNull PipelineStatus targetState
                 return;
             }
 
+            if (PipelineStatus.FAILING.equals(current) && targetState.isEndState()) {
+                targetState = PipelineStatus.FAILED;
+            }
```

Comment:
cc @Hisoka-X @ic4y

Reply:
It feels like a remedial measure. After the pipeline FAILING stage, it is not FAILED. It's weird, maybe we need to know the root cause.

Reply:
> It feels like a remedial measure. After the pipeline FAILING stage, it is not FAILED. It's weird, maybe we need to know the root cause.

Because the pipeline state must update by tasks, If the pipeline can not get enough slot, the pipeline state because `Failing` and then cancel all tasks in this pipeline. Because the tasks never run, so the tasks will complete with `CANCELED`. If all task complete with `CANCELED`, the pipeline future will complete with `CANCELED`. But this job need end with `FAILED`.

Reply:
done. 
Update the code and judge status when all tasks are completed.